src/components/withNavigationContext.js

```diff
@@ -0,0 +1,68 @@
+import React, {Component} from 'react';
+import PropTypes from 'prop-types';
+import {NavigationContext} from '@react-navigation/native';
+import getComponentDisplayName from '../libs/getComponentDisplayName';
+
+const navigationContextPropTypes = {
+    // Whether this screen is focused
+    isScreenFocused: PropTypes.bool.isRequired,
+};
+
+export default function (WrappedComponent) {
+    class withNavigationContext extends Component {
+        constructor(props) {
+            super(props);
+            this.state = {
+                isScreenFocused: false,
+            };
+        }
+
+        componentDidMount() {
+            this.unsubscribeFocus = this.context.addListener('focus', () => this.setState({
+                isScreenFocused: true,
+            }));
+            this.unsubscribeBlur = this.context.addListener('blur', () => this.setState({
+                isScreenFocused: false,
```

Comment:
I was trying to locate the docs for focus and blur to make sure I understand how this listener is invoked. Is this coming from [appState](https://reactnative.dev/docs/appstate#blur)?

Reply:
No. This `blur` event is coming from react-navigation navigation context [here](https://reactnavigation.org/docs/navigation-events/)
https://github.com/Expensify/Expensify.cash/blob/c4f64e56e764c799c736b19bf87e571cda5bda71/src/components/withNavigationContext.js#L61

Reply:
Got it, thanks for sharing 👍